datafusion/physical-expr/src/analysis.rs

```diff
@@ -64,11 +65,32 @@ impl AnalysisContext {
     ) -> Self {
         let mut column_boundaries = vec![];
         for (idx, stats) in statistics.iter().enumerate() {
-            column_boundaries.push(ExprBoundaries::from_column(
```

Comment:
This previous formulation seems clearer / more encapsulated / reusable than the change. 

Could we possibly update `ExprBoundaries::from_column` to take a `&Field` parameter and move IntervalBound construction there?